src/axom/sidre/core/Group.hpp

```diff
@@ -1308,6 +1314,30 @@ class Group
    */
   Group* deepCopyGroup(const Group* srcGroup, int allocID = INVALID_ALLOCATOR_ID);
 
+  /*!
+   * \brief Create a deep copy of Group hierarchy rooted at given Group
```

Comment:
Is there a check in this method to make sure that this group and `srcGroup` are different? If so, then I think that should be noted as it may help prevent user error. If not, then it probably should because the result could be pretty weird.